frontend/src/store/playbackManager.ts

```diff
@@ -284,22 +312,35 @@ class PlaybackManagerStore {
     }
   }
 
-  public get currentItemAssParsedSubtitleTracks(): PlaybackTrack[] {
-    const subs = this.currentItemParsedSubtitleTracks;
+  /**
+   * Filters the native subtitles
+   *
+   * As our profile requires either SSA or VTT, if it's not SSA it'll be VTT.
+   * This is done this way as server sends as "Codec" the initial value of the track, so it can be webvtt, subrip, srt...
+   * This is easier to filter out the SSA subs
+   */
+  public get currentItemVttParsedSubtitleTracks(): PlaybackExternalTrack[] {
+    return (this.currentItemParsedSubtitleTracks?.filter(
+      (sub) => sub.codec !== 'ass' && sub.codec !== 'ssa' && sub.src
+    ) || []) as PlaybackExternalTrack[];
```

Comment:
That's a technical you maybe could help. 

`this.currentItemParsedSubtitleTracks` is a `PlaybackTrack[]` and `PlaybackExternalTrack` is roughly the same item, with a few changes and especially a `src: string` and not `src?: string` as in `PlaybackTrack`.

As you see, with the filter I only keep defined `src` properties, but if I don't put the `as` cast, the linter thinks `src` can be undefined.

Reply:
Apologies, I missed this mention earlier. I think this is a good use for a type predicate, the `is` keyword, in typescript.
These aren't actually much better than an `as` cast since they still override the typing. However, I think their use makes the intent of the code much more understandable. Especially when used with array filters.

In this case we can make the filter be a type predicate. But for completeness we should also make sure that `sub.codec` is explicitly defined.
```
return (this.currentItemParsedSubtitleTracks?.filter(
      (sub): sub is PlaybackExternalTrack => sub.src && sub.codec && sub.codec !== 'ass' && sub.codec !== 'ssa'
    ) ?? []);
```

Reply:
Thanks a lot, it's corrected!